### ISSUE TITLE
Middlewares: WrapAllExcept

### ISSUE THREAD

**Issue by adamryman** [metaverse/truss#158]
The ability to exclude `Status` and `Ping` endpoints from being wrapped in middlewares has been requested. Specifically, to wrap all endpoints except `Status` and `Ping` with authentication middleware for `opsgenie`. 

Proposal: 
```
func (*e Endpoints) WrapAllExcept(middlware endpoint.Middleware, except ...endpoint.Endpoint) {
```

This function would allow you to wrap all endpoints in a middleware and specify the ones you do not want.

```
in.WrapAllExcept(authMiddleware, in.Status, in.Ping)
```

**Comment by adamryman** [comment]
Or `WrapAll` could be modified from

```
func (e *Endpoints) WrapAll(middlewares ...endpoint.Middleware) {
```
To
```
func (e *Endpoints) WrapAll(middleware endpoint.Middleware, except ...endpoint.Endpoint) {
```

And to wrap all endpoints in multiple middlewares, `WrapAll` just needs to be called multiples times.

```
in.WrapAll(authMiddleware, in.Status, in.Ping)
in.WrapAll(loggingMiddleware)
in.WrapAll(metricsMiddleware)
```

**Comment by zaquestion** [comment]
Infavor of making modifying `WrapAll`, We could make it an array of `endpoint.Middleware`

**Comment by hasIan** [comment]
I also think there's merit for auto-generating Status and Ping handlers as part of Truss generation. I mean, we're generating services that are intended to sit behind a load balancer; when aren't you going to want a status and/or ping handler is, the answer is, "Never. You will never not want these things." We have them listed in every `service.proto` we have made.

I can see an argument for `Status` to be something different, but I can imagine a generalized response that indicates a general status and includes an optional list of messages to clarify why a status may not be `STELLAR` or otherwise indicating that the service is free of trouble.

**Comment by hasIan** [comment]
Note that I still think a `WrapAllExcept` or whatnot is a good idea, even considering my most recent comment.

**Comment by adamryman** [comment]
Implemented in #164
